### DISCUSSION TITLE
Why CustomDump has dependency on xctest-dynamic-overlay

### DISCUSSION THREAD

**Discussion by InderKumarRathore** [pointfreeco/swift-custom-dump#90] [Q&A] [answered]
https://github.com/pointfreeco/swift-custom-dump/blob/e1a5818edeeaa5855365ff1ff21f8c444752f3d7/Package.swift#L19

```
  dependencies: [
    .package(url: "https://github.com/pointfreeco/xctest-dynamic-overlay", from: "0.2.0")
  ]
```

**Comment by stephencelis** [comment, marked as answer]
@InderKumarRathore One of Custom Dump's main tools is [`XCTAssertNoDifference`](https://github.com/pointfreeco/swift-custom-dump#xctassertnodifference), which provides a nice, human-readable version of `XCTAssertEqual`. It requires XCTest Dynamic Overlay to define test tools inside a library target. You can learn more about the library here: https://github.com/pointfreeco/xctest-dynamic-overlay

It's a very lightweight interface to XCTest, though, so it should compile quite fast and not be a burdensome dependency to bring in. Do you have any concerns?

**Comment by stephencelis** [comment]
Also, I'm going to convert this to a discussion since we use issues for bug tracking and this is more of a Q&A.
